app/src/main/java/com/tristanwiley/chatse/database/UserRepository.kt

```diff
@@ -0,0 +1,67 @@
+package com.tristanwiley.chatse.database
+
+import io.realm.Realm
+import io.realm.RealmObject
+import io.realm.annotations.PrimaryKey
+import io.realm.annotations.RealmClass
+
+/**
+ * Created by jaroslawmichalik on 11.10.2017
+ */
+class UserRepository : Repository<String, User> {
+
+    val myId : String = "1"
+
+    val toPojoMapper: Mapper<RealmUser, User> = RealmUserMapper()
+    val toRealmMapper: Mapper<User, RealmUser> = UserRealmMapper()
+
+    override fun findAll(): List<User> = Realm.getDefaultInstance()
+            .use { return it.where(RealmUser::class.java).findAll().map(toPojoMapper::map) }
+
+    override fun findOne(id: String): User? = Realm.getDefaultInstance()
+            .use {
+                return it.where(RealmUser::class.java)
+                        .equalTo("me", "1")
+                        .findFirst()?.let(toPojoMapper::map)
+            }
+
+    override fun findAllBy(spec: Specification) = listOf<User>()
+
+    override fun insert(t: User) = toRealmMapper.map(t).let { t1 ->
+        Realm.getDefaultInstance().use {
+            it.executeTransaction { realm -> realm.insertOrUpdate(t1) }
+        }
+    }
+
+    override fun delete(id: String) = Realm.getDefaultInstance().use {
+        it.executeTransaction { realm -> realm.where(RealmUser::class.java).equalTo("id", id).findFirst().deleteFromRealm() }
+    }
+
+    override fun deleteAll() = Realm.getDefaultInstance().use {
+        it.executeTransaction { realm -> realm.delete(RealmUser::class.java) }
+    }
+
+}
+
+class RealmUserMapper : Mapper<RealmUser, User> {
+    override fun map(f: RealmUser): User = User(f.soid, f.seid, f.name, f.email)
+
+}
+
+class UserRealmMapper : Mapper<User, RealmUser> {
+    override fun map(f: User): RealmUser = RealmUser(f.soid, f.seID, f.name, f.email)
+}
+
+
+data class User(var soid: Int = -1, var seID: Int = -1, var name: String = "", var email: String = "")
+
+@RealmClass
+open class RealmUser(
```

Comment:
What's the purpose of two classes here instead of just one that is a RealmObject? Not requiring we change it, but curious to understand your design decision.

Reply:
I think it's better to keep Realm separated from network and view models. E.X you want to switch to RoomDB or raw SqLite - you dont change RealmUser everywhere in your codebase, but just change repository implementation and object mappers.

Views and other non-repository components should not have access to RealmObject methods.

Especially in Realm there can be some strange behaviours if you manipulate object outside the transaction. In this solution, we use RealmUser (which extends RealmObject) only in repository impl, and only inside the transactions.

Reply:
Why would you use realm if you're only using unmanaged data? You're intentionally avoiding one of its best features. Might as well use plain sqlite